tests/TestCase.php

```diff
@@ -46,13 +46,59 @@ abstract class TestCase extends PHPTestCase
 
     protected $rootDir;
 
+    private $catchAllErrorHandler;
+
+    function __construct() {
+        parent::__construct();
+
+        // PHP does not implement setting a class property to an anonymous function,
+        // so we have to do it in the constructor.
+        $this->catchAllErrorHandler = function ($typeNumber, $message, $file, $lineNumber) {
+            restore_error_handler();
+            $this->fail(
+                sprintf('"%s" in %s:%d',
+                    $message,
+                    $file,
+                    $lineNumber
+                )
+            );
+        };
+    }
+
     public function setUp()
     {
         parent::setUp();
 
         $this->rootDir = __DIR__.'/..';
     }
 
+    public function tearDown() {
+        // if we changed the error handler using catchAllErrors(), reset it now
+        // this is a workaround for PHP providing a way to directly get the current error handler,
+        // by temporarily setting it to var_dump and immediately resetting it.
+        // This is also used in PHPUnit's phpunit-bridge, see
+        // https://github.com/symfony/phpunit-bridge/blob/b6c713f26fd7ec6d0c77934e812f8c3d181e2fd2/DeprecationErrorHandler.php#L183-L184
+        $currentErrorHandler = set_error_handler('var_dump');
+        restore_error_handler();
+        if($currentErrorHandler === [$this, 'catchAllErrorHandler']) {
+            restore_error_handler();
+        }
```

Comment:
Why call `restore_error_handler` a second time?

Reply:
The first one just resets the error handler of the workaround that sets it to `var_dump`, the second reset is only executed if the current error handler (which we got via the workaround) is the error handler set via the class itself. This was made to make sure that we do not reset the PHPUnit error handler by accident.

Anayway, I have removed that code from the branch now!

Reply:
> The first one just resets the error handler of the workaround that sets it to var_dump, the second reset is only executed if the current error handler (which we got via the workaround) is the error handler set via the class itself. This was made to make sure that we do not reset the PHPUnit error handler by accident.

Ah OK. I just wanted to make sure that only valid code remained. Your explanation makes sense.

> Anayway, I have removed that code from the branch now!

OK, if the code still works as intended.